### ISSUE TITLE
Predicate support for connecting to Metamask

### ISSUE THREAD

**Issue by luizstacio** [FuelLabs/fuels-ts#493]
Today the way we build predicates enables users to spend coins from a certain predicate when matching the script requirements. The idea is to have a built-in predicate that enables users to operate with Metamask as a way to sign transactions on Fuel, without the need to have a Fuel Wallet.

The ideal implementation would be a Wallet that uses a Predicate as the underneath signers replacing the need for private keys.

Use case;
- When the application calls the sign method from Predicate Wallet, Predicate will use a MetaMask SDK to request a signature from the user. When it returns, we should treat and add this to the TX  and continue the current flow.

**Comment by danielbate** [comment]
Agreed this should be done via the wallet connector with examples / documentation
